plugins/hypervisors/xenserver/src/main/java/org/apache/cloudstack/storage/motion/XenServerStorageMotionStrategy.java

```diff
@@ -326,7 +327,13 @@ private Answer migrateVmWithVolumesAcrossCluster(VMInstanceVO vm, VirtualMachine
                     volumeToStorageUuid.add(new Pair<>(volumeTo, iqn));
                 }
                 else {
-                    volumeToStorageUuid.add(new Pair<>(volumeTo, ((StoragePool)entry.getValue()).getUuid()));
+                    StoragePool pool = (StoragePool)entry.getValue();
+                    String srNameLabel = pool.getUuid();
+                    if (Storage.StoragePoolType.PreSetup.equals(pool.getPoolType()) &&
+                            !pool.getPath().contains(pool.getUuid())) {
+                        srNameLabel = pool.getPath().replaceFirst("/", "");
+                    }
```

Comment:
```suggestion
                    Sting srNameLabel = CitrixHelper.getSRNameLabel(pool.getUuid(), pool.getPoolType(), pool.getPath())
```
or if need be put the usitlity method elsewere, but make sure it only exists once!?